### ISSUE TITLE
Convert JdbcDataflowTaskExecutionMetadataDao to use Spring Data repositories

### ISSUE THREAD

**Issue by mminella** [spring-cloud/spring-cloud-dataflow#3560]
To get an MVP out quickly, the `JdbcDataflowTaskExecutionMetadataDao` class was implemented using JDBC.  However, all the other DAOs in Spring Cloud Data Flow have been converted to Spring Data JPA.  This DAO should be migrated as well.

**Comment by siddhantsorann** [comment]
@sabbyanandan @mminella I'd like to pick this one up if it's still up for grabs.

**Comment by sabbyanandan** [comment]
@siddhantsorann: That'd be a lofty goal, but feel free to give it a go. 😄

**Comment by siddhantsorann** [comment]
@sabbyanandan: Haha let me have a look then. Can you also refer me to any other class which was migrated to Spring Data JPA?

**Comment by siddhantsorann** [comment]
From what I gather we will have to create an entity as well for TaskExecutionMetadata and implement a JPA Repo on that?

**Comment by mminella** [comment]
The challenge is that most of the related entities are not JPA mapped. Nothing that SCDF gets from Spring Cloud Task is for example.

**Comment by siddhantsorann** [comment]
Right, I can see that Task_execution_metadata has a reference to task_execution which is also not a JPA Entity. How would you suggest we go about this in that case?

**Comment by siddhantsorann** [comment]
@mminella So the entity task_execution_metadata will be the first entity from scdf to be migrated to JPA from DAO? I can start building an entity and repo for it in that case.

**Comment by mminella** [comment]
It's been my experience that doing half JPA and half traditional JDBC causes data consistency issues between the two solutions (given caching/etc) so it's best to do an all or nothing approach which is where the complexity of this issue comes from.

**Comment by siddhantsorann** [comment]
@mminella That definitely makes sense. I would also suggest that we should look at moving all entities to JPA. I don't mind having a discussion and planning it out. Would still be eager to pick up such a migration.
